lib/pure/asynchttpserver.nim

```diff
@@ -91,46 +91,34 @@ export httpcore except parseHeader
 
 const
   maxLine = 8*1024
-
-when (NimMajor, NimMinor) >= (1, 1):
-  const
-    chunkSize = 8*1024 ## This seems perfectly reasonable for default chunkSize.
-
-  type
-    Request* = object
-      client*: AsyncSocket # TODO: Separate this into a Response object?
-      reqMethod*: HttpMethod
-      headers*: HttpHeaders
-      protocol*: tuple[orig: string, major, minor: int]
-      url*: Uri
-      hostname*: string    ## The hostname of the client that made the request.
-      body*: string # For future removal
-      bodyStream*: FutureStream[string]
-else:
-  type
-    Request* = object
-      client*: AsyncSocket # TODO: Separate this into a Response object?
-      reqMethod*: HttpMethod
-      headers*: HttpHeaders
-      protocol*: tuple[orig: string, major, minor: int]
-      url*: Uri
-      hostname*: string    ## The hostname of the client that made the request.
-      body*: string
+  chunkSize = 8*1024 ## This seems perfectly reasonable for default chunkSize.
 
 type
+  Request* = object
+    client*: AsyncSocket # TODO: Separate this into a Response object?
+    reqMethod*: HttpMethod
+    headers*: HttpHeaders
+    protocol*: tuple[orig: string, major, minor: int]
+    url*: Uri
+    hostname*: string    ## The hostname of the client that made the request.
+    body*: string
+    bodyStream*: FutureStream[string]
```

Comment:
Make it a private field please and add accessor procs for NimVersion >= (1, 1).

Reply:
@Araq what is the purpose of these `NimVersion >= (1,1)` annotations? Are you intending to reuse the newest stdlib for older versions of Nim?

Reply:
It's so that when we release Nim 1.2 it can compile every 1.0 project via ``--useVersion:1.0``. Most code won't need --useVersion, but it's for an extra layer of safety since we want to maintain v1.0 for a couple of years.

Reply:
> Most code won't need --useVersion, but it's for an extra layer of safety since we want to maintain v1.0 for a couple of years.

I'm going to challenge you here, because I think this is a bad idea. This code is already becoming really convoluted: we're duplicating code with small changes here and there. Who is going to maintain each branch? What is the point of this at all?

If you want to compile a project as if it's v1.0 then why not just use Nim 1.0?

Reply:
> Who is going to maintain each branch?

Well certainly not you anyway.

> What is the point of this at all?
> If you want to compile a project as if it's v1.0 then why not just use Nim 1.0?

Because we cannot backport bugfixes into 1.0 forever and the competitors have flags like ``-std=c++11`` and it works.